util/network-devp2p/src/disk.rs

```diff
@@ -0,0 +1,94 @@
+use log::*;
+use parity_crypto::publickey::Secret;
+use parity_path::restrict_permissions_owner;
+use std::fmt::Debug;
+use std::fs;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+use std::str::FromStr;
+
+pub trait DiskEntity: FromStr {
+	const PATH: &'static str;
+	const DESC: &'static str;
+
+	fn to_repr(&self) -> String;
+}
+
+impl DiskEntity for Secret {
+	const PATH: &'static str = "key";
+	const DESC: &'static str = "key file";
+
+	fn to_repr(&self) -> String {
+		self.to_hex()
+	}
+}
+
+pub fn save<E: DiskEntity>(path: &Path, entity: &E) {
+	let mut path_buf = PathBuf::from(path);
+	if let Err(e) = fs::create_dir_all(path_buf.as_path()) {
+		warn!("Error creating {} directory: {:?}", E::DESC, e);
+		return;
+	};
+	path_buf.push(E::PATH);
+	let path = path_buf.as_path();
+	let mut file = match fs::File::create(&path) {
+		Ok(file) => file,
+		Err(e) => {
+			warn!("Error creating {}: {:?}", E::DESC, e);
+			return;
+		}
+	};
+	if let Err(e) = restrict_permissions_owner(path, true, false) {
+		warn!(target: "network", "Failed to modify permissions of the file ({})", e);
```

Comment:
maybe not the right `target` here?

Reply:
Moved from `host.rs` verbatim, so no idea.